sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/GetCompletionsStreamSample.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.ai.openai.models.Choice;
+import com.azure.ai.openai.models.Completions;
+import com.azure.ai.openai.models.CompletionsOptions;
+import com.azure.ai.openai.models.CompletionsUsage;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.util.IterableStream;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sample demonstrates how to get completions as a stream for the provided input prompts. Completions support a wide variety of
+ * tasks and generate text that continues from or "completes" provided prompt data.
+ */
+public class GetCompletionsStreamSample {
+    /**
+     * Runs the sample algorithm and demonstrates how to get completions for the provided input prompts.
+     * Completions support a wide variety of tasks and generate text that continues from or "completes" provided
+     * prompt data.
+     *
+     * @param args Unused. Arguments to the program.
+     */
+    public static void main(String[] args) {
+        String azureOpenaiKey = "{azure-open-ai-key}";
+        String endpoint = "{azure-open-ai-endpoint}";
+        String deploymentOrModelId = "{azure-open-ai-deployment-model-id}";
+
+        OpenAIClient client = new OpenAIClientBuilder()
+            .endpoint(endpoint)
+            .credential(new AzureKeyCredential(azureOpenaiKey))
+            .buildClient();
+
+        List<String> prompt = new ArrayList<>();
+        prompt.add("Say this is a test");
+        IterableStream<Completions> completionsStream = client.getCompletionsStream(deploymentOrModelId,
+            new CompletionsOptions(prompt).setMaxTokens(1000).setStream(true));
+
+        completionsStream.forEach(completions -> {
```

Comment:
Is the lambda passed to `forEach` subscribing to the `Flux` ? Meaning, that if we were to translate this into a test, would we be able to embed assertions within it?

Reply:
Yes, forEach() will subscribe to the underlying flux and yes, you can have assertions in this lambda for testing.